crates/ruma-client-api/src/sync/sync_events/v4.rs

```diff
@@ -264,6 +275,12 @@ pub struct RoomSubscription {
     pub timeline_limit: Option<UInt>,
 }
 
+/// Old Rooms include information
+pub type IncludeOldRooms = RoomDetailsConfig;
+
+/// The RoomSubscriptions of the SlidingSync Request
+pub type RoomSubscription = RoomDetailsConfig;
```

Comment:
What are these type aliases good for?

Reply:
Right now, the spec has the same fields for these separate subfields. this patch is unifying them into on struct that can be reused. but as we don't know if these will stay in sync in the future, these aliases are already in place to allow to stick to the same types even if they diverge again in the future.

Reply:
I'd prefer we copy-paste the type definition then, to also allow for forwards-compatibility between Ruma versions. As-is, you can rely on `IncludeOldRooms`, `RoomSubscription` and `RoomDetailsConfig` being the same type in Rust code.